Fall back to `eglGetProcAddress` to load EGL entry points

See:
    https://bugs.webkit.org/show_bug.cgi?id=236438


--- a/Source/WebCore/platform/graphics/OpenGLShims.cpp	2022-02-08 17:15:02.000000000 +0000
+++ b/Source/WebCore/platform/graphics/OpenGLShims.cpp	2022-02-14 10:44:48.484537641 +0000
@@ -48,16 +48,26 @@
 {
     static bool initialized = false;
     static glGetProcAddressType getProcAddressFunction = 0;
+    static glGetProcAddressType getProcAddressFunctionEGL = 0;
 
     if (!initialized) {
         getProcAddressFunction = reinterpret_cast<glGetProcAddressType>(dlsym(RTLD_DEFAULT, "glXGetProcAddress"));
         if (!getProcAddressFunction)
             getProcAddressFunction = reinterpret_cast<glGetProcAddressType>(dlsym(RTLD_DEFAULT, "glXGetProcAddressARB"));
+
+        getProcAddressFunctionEGL = reinterpret_cast<glGetProcAddressType>(dlsym(RTLD_DEFAULT, "eglGetProcAddress"));
+
+        initialized = true;
     }
 
     if (!getProcAddressFunction)
         return dlsym(RTLD_DEFAULT, procName);
-    return getProcAddressFunction(procName);
+
+    auto result = getProcAddressFunction(procName);
+    if(!result && getProcAddressFunctionEGL)
+        result = getProcAddressFunctionEGL(procName);
+
+    return result;
 }
 #endif
 
